packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php

```diff
@@ -37,14 +37,20 @@ public function getParentClassMethod(ClassMethod $classMethod): ?MethodReflectio
 
         /** @var string $methodName */
         $methodName = $this->nodeNameResolver->getName($classMethod);
-        $parentClassReflections = array_merge($classReflection->getParents(), $classReflection->getInterfaces());
```

Comment:
before this change the whole type hierarchy had to be reflected which can be costly.

with this PR we only reflect what on level at a time